chemicaltoolbox/autodock_vina/docking/docking.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Galaxy Tools Project
+# Tool implementation for virtual screening
+# @authors: Léo Biscassi - leo.biscassi@gmail.com
+#           Rodrigo Faccioli - rodrigo.faccioli@gmail.com
+
+import argparse
+import os
+import shutil
+from subprocess import Popen, PIPE
+
+
+def load_parameters():
+    parser = argparse.ArgumentParser(description="script for run docking")
+    # Parameter for ligand
+    parser.add_argument("-l", action="store", dest="ligand", required=True)
+    # Parameter for receptor
+    parser.add_argument("-r", action="store", dest="receptor", required=True)
+    # Parameter for box of active site
+    parser.add_argument("-b", action="store", dest="box", required=True)
+    # Parameter for path output
+    parser.add_argument("-p", action="store",
+                        dest="path_output", required=True)
+    # Parameter for structure file
+    parser.add_argument("-f", action="store",
+                        dest="file_output1", required=True)
+    # Parameter for log file
+    parser.add_argument("-g", action="store",
+                        dest="file_output2", required=True)
+
+    arguments = parser.parse_args()
+
+    return arguments
+
+
+def load_settings():
+    vina = os.environ["VINA"]
+    return vina
+
+
+def check_autodock_vina(autodock_vina):
+    if not os.path.isfile(autodock_vina):
+        raise EnvironmentError("*** ERROR: Autodock Vina not found! ***")
+
+
+def check_pdbqt_path(path_pdbqt):
+    if not os.path.exists(path_pdbqt):
+        os.makedirs(path_pdbqt)
+
+
+def convert_dat_for_pdbqt(path_dat_file, path_pdbqt):
+    if not os.path.isfile(path_dat_file):
+        raise EnvironmentError("*** ERROR: dataset not found! ***")
+
+    shutil.copy(path_dat_file, path_pdbqt)
+
+    for data_file in os.listdir(path_pdbqt):
+        if data_file.endswith(".dat"):
+            new_file = data_file.split(".")[0] + ".pdbqt"
```

Comment:
Is this trick the main reason why you are using this extra python file?
You could do this all in your wrapper in cheetah code. For example like this one: `https://github.com/bgruening/galaxytools/blob/master/tools/diamond/diamond.xml#L10`

This would save you one file and all the argparse stuff, which is usually a huge overhead.